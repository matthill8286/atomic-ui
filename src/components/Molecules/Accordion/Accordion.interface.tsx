import { FoldableAcceptedBreakpoints } from '@/components/Atoms/Foldable/Foldable.interface'
import { BoxDimensions, ThemeColors, ThemeFontSizes, ThemeFontWeights } from '@/types/theme'

export interface AccordionChangeEvent {
  index: number
  eventType: 'HIDE' | 'SHOW'
}

type accordionEntryChangeHandler = (id: string) => void

export interface AccordionProps {
  className?: string
  entries: AccordionEntryObject[]
  isLarge?: boolean
  isUnfoldable?: boolean /* allows multiple entries to be open at the same time */
  isExpandedFrom?:
    | ''
    | 'sm'
    | 'md'
    | 'lg'
    | 'xl'
    | 'xxl' /* is in expanded mode from given size upwards but on smaller sizes looks the same as default */
  openedIndex?: number
  onChange?: (event: AccordionChangeEvent) => void
  customLabelHeading?: LabelHeadingProps
  labelPadding?: BoxDimensions
  entryPadding?: BoxDimensions
  withIconsOnRight?: boolean
  withCustomIcon?: boolean
  customIcon?: JSX.Element | React.ReactNode
}

export interface AccordionRefObject {
  openEntryById: (id: string) => void
}

export interface AccordionEntryObject {
  details: React.ReactNode
  id?: string
  isOpenInitially?: boolean
  noBorderTop?: boolean
  title?: React.ReactNode
}

export interface AccordionEntryProps {
  ref?: React.Ref<HTMLElement>
  details: React.ReactNode
  isLarge?: boolean
  id: string
  isOpen?: boolean
  looksOpenInitiallyFromBreakpoint?: FoldableAcceptedBreakpoints
  onChange: accordionEntryChangeHandler
  title: React.ReactNode
  customLabelHeading?: LabelHeadingProps
  labelPadding?: BoxDimensions
  entryPadding?: BoxDimensions
  noBorderTop?: boolean
  withIconsOnRight?: boolean
  withCustomIcon?: boolean
  customIcon?: JSX.Element | React.ReactNode
}

export interface LabelHeadingProps {
  tag?: string
  size?: ThemeFontSizes
  color?: ThemeColors
  bold?: ThemeFontWeights
  margin?: BoxDimensions
}

export interface OpenCloseAction {
  type: 'open' | 'close'
  id: string
  shouldCloseOthers?: boolean
}

export interface OpenBatchAction {
  type: 'openBatch'
  entries: AccordionEntryObject[]
}

export interface CheckNewEntriesAction {
  type: 'checkNewEntries'
  entries: AccordionEntryObject[]
}

export type ActionType = OpenCloseAction | OpenBatchAction | CheckNewEntriesAction

export interface StyledAccordionEntryProps {
  noBorder: boolean
}

export interface StyledLabelProps {
  isLarge: boolean
  withIconsOnRight: boolean
  padding?: BoxDimensions
  withCustomIcon?: boolean
  customIcon?: JSX.Element | React.ReactNode
}

export interface StyledLabelTextProps {
  withoutIconSpace: boolean
  isLarge: boolean
}

export interface StyledEntryContentProps {
  isLarge: boolean
  padding?: BoxDimensions
}
