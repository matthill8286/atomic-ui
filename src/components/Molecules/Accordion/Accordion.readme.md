# Accordion Molecule

The accordion component lets users show and hide sections of detailed content on a page. Accordions help break down complex and detailled informations into small pieces which the user can expand at their own pace. 
Accordions hide content, so the headline need to be clear.

It has a "large" variant with a larger label font. And one can specify a breakpoint (xs-xxl) and from that breakpoint upwards the accordion entries are statically open. Useful for compact layouts on small screens and showing constantly all information on larger screens.

To have a accordion entry without a border set `noBorderTop: true` otherwise all entries will have `borderTop`

The Accordion can also set the entry to be initiallyOpen by setting `isOpenInitially: true` on the entry that needs be open.  
