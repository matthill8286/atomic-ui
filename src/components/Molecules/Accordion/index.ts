export * from './Accordion'
export * from './AccordionEntry'
import {
  AccordionEntryObject as EntryObject,
  AccordionRefObject as RefObject,
} from './Accordion.interface'
export type AccordionEntryObject = EntryObject
export type AccordionRefObject = RefObject
