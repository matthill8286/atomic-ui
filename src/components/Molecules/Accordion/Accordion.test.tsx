import React from 'react'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { Accordion } from './Accordion'

const ENTRIES_1 = [
  {
    title: 'Produktbewertungen',
    id: 'ratings',
    details: `Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam`,
  },
  {
    title: 'Technische Daten',
    id: 'technicals',
    details: `Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam`,
    isOpenInitially: true,
  },
  {
    title: 'Produktbeschreibung',
    id: 'description',
    details: `At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam`,
  },
  {
    title: 'ProduktbeschreibungProduktbeschreibungProduktbeschreibungProduktbeschreibung',
    id: 'production-2',
    details: `At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam`,
  },
]

describe('Accordion', () => {
  it('renders accordion with initial open entry correctly', () => {
    const tree = renderWithTheme(<Accordion entries={ENTRIES_1} />)
    expect(tree).toMatchSnapshot()
  })
  it('can render all entries open dependant on breakpoint', () => {
    const tree = renderWithTheme(<Accordion isExpandedFrom="sm" entries={ENTRIES_1} />)
    expect(tree).toMatchSnapshot()
  })
  it('has a large variant', () => {
    const tree = renderWithTheme(<Accordion isLarge entries={ENTRIES_1} />)
    expect(tree).toMatchSnapshot()
  })

  it('should not render heading h2 element when title undefined', () => {
    const wrapper = mountWithTheme(
      <Accordion isLarge entries={ENTRIES_1.map(e => ({ ...e, title: undefined }))} />
    )
    expect(wrapper.find('h2')).toHaveLength(0)
  })
})
