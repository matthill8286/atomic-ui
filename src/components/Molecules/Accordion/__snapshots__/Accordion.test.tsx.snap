// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion can render all entries open dependant on breakpoint 1`] = `
.c6 {
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 svg {
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  fill: currentColor;
}

.c2 svg g,
.c2 svg p {
  fill: currentColor;
}

.c4 {
  margin: 0 16px;
  color: #666666;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  overflow: hidden;
  max-height: calc(24px * 2);
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

.c7 {
  color: #666666;
  font-size: 14px;
  line-height: 20px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  border-top: 1px solid #E5E5E5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-family: OswaldLight,Arial,sans-serif;
  padding: 16px 0;
}

.c5 {
  word-break: break-word;
  max-height: calc(24px * 4);
  -webkit-line-clamp: 4;
}

.c3 svg {
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
}

.c8 {
  padding: 16px;
}

.c8 font-weight:function (_ref6) {
  -webkit-var: theme = _ref6.theme;
  -moz-var: theme = _ref6.theme;
  -ms-var: theme = _ref6.theme;
  var: theme = _ref6.theme;
  return: theme.font.weight.regular;
}

@media (min-width:512px) {

}

@media (max-width:512px) {
  .c1 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<section>
  <section
    className="c0 ewb-accordion-open"
    id="ratings"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          data-name="Layer 1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.5 13a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h13a.5.5 0 01.5.5v1a.5.5 0 01-.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Produktbewertungen
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": "auto",
          "transition": "",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-open"
    id="technicals"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          data-name="Layer 1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.5 13a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h13a.5.5 0 01.5.5v1a.5.5 0 01-.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Technische Daten
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": "auto",
          "transition": "",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-open"
    id="description"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          data-name="Layer 1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.5 13a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h13a.5.5 0 01.5.5v1a.5.5 0 01-.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Produktbeschreibung
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": "auto",
          "transition": "",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-open"
    id="production-2"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          data-name="Layer 1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.5 13a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h13a.5.5 0 01.5.5v1a.5.5 0 01-.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        ProduktbeschreibungProduktbeschreibungProduktbeschreibungProduktbeschreibung
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": "auto",
          "transition": "",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`Accordion has a large variant 1`] = `
.c6 {
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 svg {
  width: 28px;
  height: 28px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  fill: currentColor;
}

.c2 svg g,
.c2 svg p {
  fill: currentColor;
}

.c4 {
  margin: 0 16px;
  color: #666666;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  overflow: hidden;
  max-height: calc(24px * 2);
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

.c7 {
  color: #666666;
  font-size: 14px;
  line-height: 20px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  border-top: 1px solid #E5E5E5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-family: OswaldLight,Arial,sans-serif;
  padding: 24px 0;
}

.c5 {
  word-break: break-word;
  max-height: calc(24px * 4);
  -webkit-line-clamp: 4;
}

.c3 svg {
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
}

.c8 {
  padding: 24px;
}

.c8 font-weight:function (_ref6) {
  -webkit-var: theme = _ref6.theme;
  -moz-var: theme = _ref6.theme;
  -ms-var: theme = _ref6.theme;
  var: theme = _ref6.theme;
  return: theme.font.weight.regular;
}

@media (max-width:512px) {
  .c1 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<section>
  <section
    className="c0 ewb-accordion-closed"
    id="ratings"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={28}
        rotate={90}
        width={28}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 11.5v1a.5.5 0 01-.5.5H13v5.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V13H5.5a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5H11V5.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V11h5.5a.5.5 0 01.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Produktbewertungen
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": 0,
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-open"
    id="technicals"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={28}
        rotate={90}
        width={28}
      >
        <svg
          data-name="Layer 1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.5 13a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h13a.5.5 0 01.5.5v1a.5.5 0 01-.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Technische Daten
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": "auto",
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-closed"
    id="description"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={28}
        rotate={90}
        width={28}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 11.5v1a.5.5 0 01-.5.5H13v5.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V13H5.5a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5H11V5.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V11h5.5a.5.5 0 01.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Produktbeschreibung
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": 0,
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-closed"
    id="production-2"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={28}
        rotate={90}
        width={28}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 11.5v1a.5.5 0 01-.5.5H13v5.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V13H5.5a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5H11V5.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V11h5.5a.5.5 0 01.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        ProduktbeschreibungProduktbeschreibungProduktbeschreibungProduktbeschreibung
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": 0,
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam
        </div>
      </div>
    </div>
  </section>
</section>
`;

exports[`Accordion renders accordion with initial open entry correctly 1`] = `
.c6 {
  overflow: hidden;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 svg {
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  fill: currentColor;
}

.c2 svg g,
.c2 svg p {
  fill: currentColor;
}

.c4 {
  margin: 0 16px;
  color: #666666;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 700;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  overflow: hidden;
  max-height: calc(24px * 2);
  -webkit-line-clamp: 2;
  display: -webkit-box;
  -webkit-box-orient: vertical;
}

.c7 {
  color: #666666;
  font-size: 14px;
  line-height: 20px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c0 {
  border-top: 1px solid #E5E5E5;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  font-family: OswaldLight,Arial,sans-serif;
  padding: 16px 0;
}

.c5 {
  word-break: break-word;
  max-height: calc(24px * 4);
  -webkit-line-clamp: 4;
}

.c3 svg {
  -webkit-transition: -webkit-transform 0.25s ease-in-out;
  -webkit-transition: transform 0.25s ease-in-out;
  transition: transform 0.25s ease-in-out;
}

.c8 {
  padding: 16px;
}

.c8 font-weight:function (_ref6) {
  -webkit-var: theme = _ref6.theme;
  -moz-var: theme = _ref6.theme;
  -ms-var: theme = _ref6.theme;
  var: theme = _ref6.theme;
  return: theme.font.weight.regular;
}

@media (max-width:512px) {
  .c1 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<section>
  <section
    className="c0 ewb-accordion-closed"
    id="ratings"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 11.5v1a.5.5 0 01-.5.5H13v5.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V13H5.5a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5H11V5.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V11h5.5a.5.5 0 01.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Produktbewertungen
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": 0,
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-open"
    id="technicals"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          data-name="Layer 1"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.5 13a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5h13a.5.5 0 01.5.5v1a.5.5 0 01-.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Technische Daten
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": "auto",
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
              nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat,
              sed diam voluptua. At vero eos et accusam et justo duo dolores et ea
              rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem
              ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur
              sadipscing elitr, sed diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-closed"
    id="description"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 11.5v1a.5.5 0 01-.5.5H13v5.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V13H5.5a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5H11V5.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V11h5.5a.5.5 0 01.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        Produktbeschreibung
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": 0,
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam
        </div>
      </div>
    </div>
  </section>
  <section
    className="c0 ewb-accordion-closed"
    id="production-2"
  >
    <div
      className="c1"
      data-test="accordion-header"
      onClick={[Function]}
    >
      <div
        className="c2 c3"
        height={24}
        rotate={90}
        width={24}
      >
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 11.5v1a.5.5 0 01-.5.5H13v5.5a.5.5 0 01-.5.5h-1a.5.5 0 01-.5-.5V13H5.5a.5.5 0 01-.5-.5v-1a.5.5 0 01.5-.5H11V5.5a.5.5 0 01.5-.5h1a.5.5 0 01.5.5V11h5.5a.5.5 0 01.5.5z"
          />
        </svg>
      </div>
      <div
        className="c4 c5"
        color="grey5"
      >
        ProduktbeschreibungProduktbeschreibungProduktbeschreibungProduktbeschreibung
      </div>
    </div>
    <div
      className="c6"
      data-test="foldable"
      onTransitionEnd={[Function]}
      style={
        Object {
          "height": 0,
          "transition": "height 300ms linear",
        }
      }
    >
      <div
        data-test="inner-foldable"
      >
        <div
          className="c7 c8"
          color="grey5"
          fontSize="xs"
        >
          At vero eos et accusam et justo duo dolores et ea rebum. Stet clita
              kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit
              amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed
              diam
        </div>
      </div>
    </div>
  </section>
</section>
`;
